sw/device/silicon_creator/lib/crypto/tests/rsa_3072_verify_functest.c

```diff
@@ -2,150 +2,64 @@
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555, 0x55555555, 0x55555555, 0x55555555, 0x55555555,
-             0x55555555}};
+  // Encode message
+  RETURN_IF_ERROR(
+      rsa_3072_encode_sha256(testvec->msg, testvec->msgLen, &encodedMessage));
 
-rom_error_t compute_constants_test(void) {
-  rsa_3072_constants_t act_constants;
+  // Precompute Montgomery constants
+  FOLD_OTBN_ERROR(rsa_3072_compute_constants(&testvec->publicKey, &constants));
 
-  // Precompute constants
-  FOLD_OTBN_ERROR(rsa_3072_compute_constants(&kPublicKey, &act_constants));
+  // Attempt to verify signature
+  FOLD_OTBN_ERROR(rsa_3072_verify(&testvec->signature, &encodedMessage,
+                                  &testvec->publicKey, &constants, &result));
 
-  // Check that RR matches expected value
-  for (int i = 0; i < kRsa3072NumWords; i++) {
-    CHECK(act_constants.rr.data[i] == kExpConstants.rr.data[i],
-          "Mismatch at word %d of R^2: expected=0x%x, actual=0x%x", i,
-          kExpConstants.rr.data[i], act_constants.rr.data[i]);
+  if (testvec->valid) {
+    CHECK(result == kHardenedBoolTrue);
+  } else {
+    CHECK(result == kHardenedBoolFalse);
   }
 
-  // Check that m0_inv matches expected value
-  for (int i = 0; i < kOtbnWideWordNumWords; i++) {
-    CHECK(act_constants.m0_inv[i] == kExpConstants.m0_inv[i],
-          "Mismatch at word %d of m0_inv: expected=0x%x, actual=0x%x", i,
-          kExpConstants.m0_inv[i], act_constants.m0_inv[i]);
-  }
-
-  return kErrorOk;
-}
-
-rom_error_t verify_test(void) {
-  hardened_bool_t result;
-
-  FOLD_OTBN_ERROR(rsa_3072_verify(&kSignature, &kMessage, &kPublicKey,
-                                  &kExpConstants, &result));
-
-  CHECK(result == kHardenedBoolTrue);
-
   return kErrorOk;
 }
 
 const test_config_t kTestConfig;
 
 bool test_main(void) {
+  rsa_3072_verify_test_vector_t testvec;
+  rom_error_t localErr;
```

Comment:
ITYM `local_err` =)

That said, you can fold this, as well as the `test_vec` variabiable, into the loop; they don't need to be predeclared.

Reply:
Ack for folding the variables into the loop, I'll do that.

As for the error type, I'm still using the ROM test infrastructure (for now; this might change soon), so that's why the error type is a ROM error. Although with these changes, I've basically inlined `EXECUTE_TEST`, which was the main reason I was using ROM error types in tests, so I think I'll just resolve this by changing to OTBN error types, which are what the verify routine returns.

Reply:
Sure